### ISSUE TITLE
🐛 `noUnusedVariables`: valid type import and use incorrectly flagged

### ISSUE THREAD

**Issue by nstepien** [biomejs/biome#606]
### Environment information

```block
CLI:
  Version:                      1.3.1
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           "v21.1.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.2.1"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    true
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### What happened?

```ts
import type { value } from './whatever';

export type Test = typeof value;
```

Biome will flag the `value` import as unused, when it is used to get its type.
In our code `value` is not a type but a function (React component), but we can still import it with `import type` as we only need its type.

### Expected result

Type import + `typeof x` should still count as used.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
The issue was recently fixed in #566. The fix will be shipped in the next version.
